wikimedia/wikipedia-ios
# Image recommendations - Time-box new feature announcement

**Phabricator:** https://phabricator.wikimedia.org/T364397

### Notes
* PR adds an early return for the announcement of Image Recommendations for dates later than Oct 4, 2024. Added all logic in a single struct for easy cleanup after.
* Also added my name to collaborators :) Thanks for the notice!

### Test Steps
1. Toggle [this boolean](https://github.com/wikimedia/wikipedia-ios/blob/08f5693881a6d0fac23e649643f673a3fb4e724e/WMF%20Framework/FeatureFlags.swift#L24) for being able to get offered the image recommendations feature
2. override [hasPresentedFeatureAnnouncementModal](https://github.com/wikimedia/wikipedia-ios/blob/f07b1f83f643bd86f65d863b07e151f3b160d428/WKData/Sources/WKData/Data%20Controllers/Image%20Recommendations/WKImageRecommendationsDataController.swift#L55) to false to ignore UserDefaults and show the announcement modal.
3. Check that modal still shows up as we are currently before the expiration date.

### Screenshots/Videos

| | 
| ------ | 
| ![featureannouncement](https://github.com/wikimedia/wikipedia-ios/assets/34898507/c95c7b83-7d64-4440-86b9-0fe268522272) |
